src/Umbraco.Core/Packaging/PackageBinaryInspector.cs

```diff
@@ -29,20 +29,24 @@ internal class PackageBinaryInspector : MarshalByRefObject
         /// </remarks>
         public static IEnumerable<string> ScanAssembliesForTypeReference<T>(IEnumerable<byte[]> assemblys, out string[] errorReport)
         {
-            var appDomain = GetTempAppDomain();
-            var type = typeof(PackageBinaryInspector);
-            try
+            // beware! when toying with domains, use a safe call context!

```

Comment:
Derives from modifying the DefaultDatabaseFactory to store the non-HttpContext database in logical call context vs. ThreadStatic, because the ThreadStatic thing does not support async, which we need for Deploy. And _then_ we have to clean the logical call context when toying with domains in tests. Makes sense to backport, backporting.